### PR TITLE
feat: mgmt-ui FE for destinations

### PR DESCRIPTION
**Pull request by albertyfwu**
mgmt-ui FE for destinations

image

image

## Test Plan

[Describe test plan here]

## Deployment instructions

[Add any special deployment instructions here]

### CODE REVIEW

File: apps/mgmt-ui/src/components/configuration/DestinationCard.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
+/* eslint-disable @typescript-eslint/no-floating-promises */
```
Comment (asdfryan) [inline apps/mgmt-ui/src/components/configuration/DestinationCard.tsx:1]: where is the floating promise?

Comment (albertyfwu) [reply apps/mgmt-ui/src/components/configuration/DestinationCard.tsx:1]: idk just copied this from another file.

File: apps/mgmt-ui/src/pages/api/internal/destinations/update.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { getOrgId } from '@/utils/org';
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { API_HOST, SG_INTERNAL_TOKEN } from '../..';
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  const result = await fetch(`${API_HOST}/internal/v1/destinations/${req.body.id}`, {
+    method: 'PUT',
+    headers: {
```
Comment (asdfryan) [inline apps/mgmt-ui/src/pages/api/internal/destinations/update.ts:8]: Can we take the opportunity to DRY this (at least the headers)

File: apps/mgmt-ui/src/components/configuration/DestinationCard.tsx (hunk 1 above)
Comment (albertyfwu) [reply apps/mgmt-ui/src/components/configuration/DestinationCard.tsx:1]: just checked. `router.push` is a promise

File: apps/mgmt-ui/src/components/configuration/DestinationDetailsPanel.tsx [hunk 3]
```diff
@@ -0,0 +1,170 @@
+/* eslint-disable @typescript-eslint/no-floating-promises */
```
Comment (asdfryan) [inline apps/mgmt-ui/src/components/configuration/DestinationDetailsPanel.tsx:1]: Where is the floating promise?

File: apps/mgmt-ui/src/utils/destinationTypeToIcon.tsx [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import PostgresIcon from '@/assets/destination_icons/postgres.png';
+import S3Icon from '@/assets/destination_icons/s3.png';
+import Image from 'next/image';
+import { ReactNode } from 'react';
+
+export default function destinationTypeToIcon(type: string, size = 25): ReactNode {
```
[the remark this answers is not in the document]
Comment (albertyfwu) [reply apps/mgmt-ui/src/utils/destinationTypeToIcon.tsx:6]: why do we want to use the same method? they are different concepts

[the remark this answers is not in the document]
Comment (asdfryan) [reply apps/mgmt-ui/src/utils/destinationTypeToIcon.tsx:6]: It looks like the code is identical if you just rename `type` to `providerName`, so I think we should just have a single mapping from company name to logo

[the remark this answers is not in the document]
Comment (albertyfwu) [reply apps/mgmt-ui/src/utils/destinationTypeToIcon.tsx:6]: ok

File: apps/mgmt-ui/src/pages/api/internal/destinations/update.ts (hunk 2 above)
Comment (albertyfwu) [reply apps/mgmt-ui/src/pages/api/internal/destinations/update.ts:8]: ok

File: apps/mgmt-ui/src/utils/destinationTypeToIcon.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (asdfryan) [reply apps/mgmt-ui/src/utils/destinationTypeToIcon.tsx:4]: Nit: can we just call companyToIcon directly

**Pull request merged**